src/models/polar.jl

```diff
@@ -0,0 +1,509 @@
+    qinj[pf.pq] .= p[nref + npq + 1:nref + 2*npq]
+
+    for bus in pf.ref
+        pinj[bus] = PS.get_power_injection(bus, vmag, vang, polar.ybus_re, polar.ybus_im)
+        qinj[bus] = PS.get_react_injection(bus, vmag, vang, polar.ybus_re, polar.ybus_im)
+    end
+
+    for bus in pf.pv
+        qinj[bus] = PS.get_react_injection(bus, vmag, vang, polar.ybus_re, polar.ybus_im)
+    end
+
+    return vmag, vang, pinj, qinj
+end
+
+function power_balance(polar::PolarForm, x, u, p; V=Float64)
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    ngen = PS.get(polar.network, PS.NumberOfGenerators())
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    # Indexing
+    ref = convert(polar.AT{Int, 1}, polar.network.ref)
+    pv = convert(polar.AT{Int, 1}, polar.network.pv)
+    pq = convert(polar.AT{Int, 1}, polar.network.pq)
+    # Network state
+    Vm, Va, pbus, qbus = get_network_state(polar, x, u, p; V=V)
+    F = similar(x)
+    fill!(F, 0.0)
+    residualFunction_polar!(F, Vm, Va,
+                                  polar.ybus_re,
+                                  polar.ybus_im,
+                                  pbus, qbus, pv, pq, nbus)
+    return F
+end
+
+# TODO: find better naming
+function init_ad_factory(polar::PolarForm{T, VT, AT}, x, u, p) where {T, VT, AT}
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    ngen = PS.get(polar.network, PS.NumberOfGenerators())
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    n_states = get(polar, NumberOfState())
+    # Indexing
+    ref = convert(polar.AT{Int, 1}, polar.network.ref)
+    pv = convert(polar.AT{Int, 1}, polar.network.pv)
+    pq = convert(polar.AT{Int, 1}, polar.network.pq)
+    # Network state
+    Vm, Va, pbus, qbus = get_network_state(polar, x, u, p)
+    F = VT(undef, n_states)
+    fill!(F, zero(T))
+    # Build the AD Jacobian structure
+    stateJacobianAD = AD.StateJacobianAD(F, Vm, Va,
+                                         polar.ybus_re, polar.ybus_im, pbus, qbus, pv, pq, ref, nbus)
+    designJacobianAD = AD.DesignJacobianAD(F, Vm, Va,
+                                           polar.ybus_re, polar.ybus_im, pbus, qbus, pv, pq, ref, nbus)
+    return stateJacobianAD, designJacobianAD
+end
+
+function jacobian(polar::PolarForm, jac::AD.AbstractJacobianAD, x, u, p)
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    ngen = PS.get(polar.network, PS.NumberOfGenerators())
+    # Indexing
+    ref = convert(polar.AT{Int, 1}, polar.network.ref)
+    pv = convert(polar.AT{Int, 1}, polar.network.pv)
+    pq = convert(polar.AT{Int, 1}, polar.network.pq)
+    # Network state
+    Vm, Va, pbus, qbus = get_network_state(polar, x, u, p)
+    AD.residualJacobianAD!(jac, residualFunction_polar!, Vm, Va,
+                           polar.ybus_re, polar.ybus_im, pbus, qbus, pv, pq, ref, nbus, TIMER)
+    return jac.J
+end
+
+function powerflow(
```

Comment:
The new `solve` function, working now directly with `PolarForm`. I think we could go further in the refactoring there